xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/vue/utilities/BaseSelect.vue

```diff
@@ -101,30 +102,51 @@ export default {
     sort: {
       type: Boolean,
       default: false,
-    }
+    },
+  },
+
+  computed: {
+    optionsValues () {
+      console.log("optionsValues", this.options.map(option => option.value));
+      console.log("this.options", this.options);
+      console.log("this", this);
+      return this.options.map(option => option.value);
+    },
+
+    selected () {
+      console.log("selected", this.selectedValues.map(value => this.options.find(option => option.value === value)));
+      console.log("this.selectedValues", this.selectedValues);
+      console.log("this", this);
+      return this.selectedValues.map(value => this.options.find(option => option.value === value));
+    },
   },
 
   methods: {
+    isValidOption (option) {
+      return this.optionsValues.includes(option.value);
+    },
+
     isSelected (option) {
-      return this.selected.includes(option);
+      return this.selectedValues.includes(option.value);
     },
 
     select (option) {
-      if (!this.options.includes(option)) { return; }
+      if (!this.isValidOption(option)) { return; }
       if (this.isSelected(option)) { return; }
-      let _selected = this.selected.slice()
-      _selected.push(option);
-      this.formatSelected(_selected);
+      console.log("select", this.select);
+      const _selected = this.multiple ? this.selectedValues.slice() : [];
+      _selected.push(option.value);
+      this.sortSelected(_selected);
       this.$emit("change", _selected);
     },
 
     deselect (option) {
-      if (!this.options.includes(option)) { return; }
-      const index = this.selected.indexOf(option);
+      if (!this.isValidOption(option)) { return; }
```

Comment:
We shouldn't  prevent the user from "fixing" a filter / constraint by removing an obsolete value.